modules/experimental/src/gpgpu/point-in-polygon/gpu-point-in-polygon.js

```diff
@@ -0,0 +1,183 @@
+import GL from '@luma.gl/constants';
+import {Buffer, Texture2D, assert} from '@luma.gl/webgl';
+import {isWebGL2} from '@luma.gl/gltools';
+import {Transform} from '@luma.gl/engine';
+import {default as textureFilterModule} from './texture-filter';
+import {POLY_TEX_VS, FILTER_VS} from './shaders';
+const TEXTURE_SIZE = 512;
+import * as Polygon from './polygon';
+
+export default class GPUPointInPolygon {
+  constructor(gl, opts = {}) {
+    this.gl = gl;
+    assert(isWebGL2(gl)); // supports WebGL2 only
+    this.textureSize = TEXTURE_SIZE;
+    this._setupResources();
+    this.update(opts);
+  }
+
+  update({polygons, textureSize} = {}) {
+    if (textureSize) {
+      this.textureSize = textureSize;
+    }
+    if (!polygons || polygons.length === 0) {
+      return;
+    }
+
+    const {vertices, indices, vertexCount, ids} = triangulatePolygons(polygons);
+    this._updateResources(vertices, indices, ids, vertexCount);
+  }
+
+  filter({positionBuffer, filterValueIndexBuffer, count}) {
+    this.filterTransform.update({
+      sourceBuffers: {
+        a_position: positionBuffer
+      },
+      feedbackBuffers: {
+        filterValueIndex: filterValueIndexBuffer
+      },
+      elementCount: count
+    });
+    const {polygonTexture, boundingBox} = this;
+
+    this.filterTransform.run({
+      moduleSettings: {boundingBox, texture: polygonTexture}
+    });
+  }
+
+  // PRIVATE
+
+  _setupResources() {
+    const {gl} = this;
+
+    // texture to render polygons to
+    this.polygonTexture = new Texture2D(gl, {
+      format: GL.RGB,
```

Comment:
Why create an RGB texture if you're only writing one value: https://github.com/uber/luma.gl/pull/1360/files#diff-a588d4a9222caddc58312b3ea2d63b3aR13

This can just be GL.RED

Reply:
`GL.RED` is WebGL2 only, intention of this feature is to support this feature in WebGL environments (which will use Texture Generation and Sampling parts, except TF) by exporting it from luma.gl, we can do that in later iterations.

Reply:
Also we need two channels,  R channel for polygonID [0, 255] and G channel for inside (1) and outside (0). I will update docs to specify support for maximum of 256 polygons. If user not interested in distinguishing polygons by ID, there can be more than 256, while id getting clamped to max value of 255.